tests/test_dsmc_handlers.py

```diff
@@ -0,0 +1,453 @@
+        dsmc_log_file = "foolog"
+        descr = "foodescr"
+        run_dir = "foodir"
+        uniq_id = str(uuid.uuid4())
+        reupload_file = "/tmp/test_reupload-{}".format(uniq_id)
+
+        exp_id = 72
+
+        class MyRunner(object):
+            def start(self, cmd, nbr_of_cores, run_dir, stdout=dsmc_log_file, stderr=dsmc_log_file):
+                self.components = cmd.split("=")
+                self.cmd = cmd
+                return exp_id
+
+        runsrv = MyRunner()
+
+        local_files = {"foo": 23, "bar": 46, "uggla": 72}
+        uploaded_files = {"foo": 23, "uggla": 15}
+        exp_upload = ['"bar"\n', '"uggla"\n']
+
+        reupload_files = helper.get_files_to_reupload(local_files, uploaded_files)
+
+        def my_tmp_file(ReuploadHelper, component):
+            return reupload_file
+
+        with mock.patch \
+                ("archive_upload.handlers.dsmc_handlers.ReuploadHelper._tmp_file",\
+                autospec=True) as mock__tmp_file:
+            mock__tmp_file.side_effect = my_tmp_file
+            res_id = helper.reupload(reupload_files, descr, dsmc_log_dir, runsrv)
+
+        self.assertEqual(res_id, exp_id)
+        self.assertEqual(runsrv.components[-1], descr)
+
+        with open(reupload_file) as f:
+            uploaded = f.readlines()
+
+        import sets
+        uploaded = sets.Set(uploaded)
+        exp_upload = sets.Set(exp_upload)
+        self.assertEqual(len(uploaded.symmetric_difference(exp_upload)), 0)
+
+    def test_compress_archive_full(self):
+        import shutil
+
+        root = self.dummy_config["path_to_archive_root"]
+        archive_path = os.path.join(root, "johanhe_test_archive")
+        original = os.path.join(root, "johanhe_test_runfolder")
+
+        shutil.rmtree(archive_path, ignore_errors=True)
+        shutil.copytree(original, archive_path)
+
+        resp = self.fetch(self.API_BASE + "/compress_archive/johanhe_test_archive", method="POST",
+                          allow_nonstandard_methods=True)
+
+        json_resp = json.loads(resp.body)
+        self.assertEqual(json_resp["state"], State.DONE)
+
+        self.assertFalse(os.path.exists(os.path.join(archive_path, "RunInfo.xml")))
+        self.assertTrue(os.path.exists(os.path.join(archive_path, "Config")))
+        self.assertTrue(os.path.exists(os.path.join(archive_path, "johanhe_test_archive.tar.gz")))
+
+        shutil.rmtree(archive_path)
+
+    def test_compress_archive_mini(self):
+        import shutil
+        root = self.dummy_config["path_to_archive_root"]
+        archive_path = os.path.join(root, "testrunfolder_archive_tmp")
+        original = os.path.join(root, "testrunfolder_archive_input")
+
+        shutil.rmtree(archive_path, ignore_errors=True)
+        shutil.copytree(original, archive_path)
+
+        resp = self.fetch(self.API_BASE + "/compress_archive/testrunfolder_archive_tmp", method="POST",
+                          allow_nonstandard_methods=True)
```

Comment:
Why is the `allow_nonstandard_methods` used here?

Reply:
Tornado requires that if you want to use a `POST` without a body.

Reply:
:+1: